devs/docs/release.rst

```diff
@@ -0,0 +1,87 @@
+======================
+Git & Release workflow
+======================
+
+- Development happens in feature branches. Everything that goes into master
+  must be reviewed.
+
+- PRs should be small but self-contained. "Dead code" (Features that aren't
+  integrated) cannot go into master.
+
+- Commits must be self-contained. Each commit must compile and tests should
+  pass. (Flaky tests happen.)
+  This is required to be able to use ``git bisect``.
+
+Release workflow
+================
+
+
+A "development cycle" ends with a feature freeze and an initial release
+candidate. This RC is based on master, but looks like a regular release. We use
+``X.Y.Z`` version numbers, without ``-RC`` suffix or similar RC indication.
+These release candidates are available in our testing repositories. They're not
+suitable for production and there is no rolling upgrade support.
+
+There will likely be several release candidates being released, each time we:
+
+ - Create a "prepare release" commit. This commit updates the version-info,
+   finalizes the release notes and is tagged.
+ - Create a "version bump" commit immediately afterwards. This re-enables the
+   ``SNAPSHOT`` flag and increases the version number.
+
+So for example::
+
+    - Prepare release 2.1.0
+    - Version bump (2.1.1-SNAPSHOT)
+    - [... fixes ...]
+    - Prepare release 2.1.1
+    - Version bump (2.1.2-SNAPSHOT)
+    - [... repeat ...]
+
+During this feature freeze we can no longer merge new features into master. We
+focus on fixes, performance improvements or other cleanups.
+
+
+At some point we'll lift the feature freeze. At this point a ``major.minor``
+release branch is created and we bump the ``minor`` version on master.
+
+
+When the number of bugs discovered in the release candidates decreases enough
+and it's considered stable, we move the release candidate into our stable
+repositories and declare it stable.
```

Comment:
would add something like `so this will be the first stable release of this ``minor`` version.